Week1/homework/app.js

```diff
@@ -1,11 +1,104 @@
 'use strict';
 {
+  /*
   const bookTitles = [
     // Replace with your own book titles
-    'harry_potter_chamber_secrets'
+    'the_nose',
+    'the_overcoat',
+    'the_collected_works',
+    'palace_of_desire',
+    'sugar_street',
+    'children_of_geblawi',
+    'the_thief_and_the_dogs',
+    'autumn_quail',
+    'a_tale_of_two_cities',
+    'the_happy_prince'
   ];
 
-
   // Replace with your own code
-  console.log(bookTitles);
+  //console.log(bookTitles);
+  
+    function generateBookList(array) {
+      const container = document.getElementById('container');
+      const listOfBooks = document.createElement('ul');
+      container.appendChild(listOfBooks);
+      for (const element of array) {
+        const book = document.createElement('li');
+        book.innerHTML = element;
+        listOfBooks.appendChild(book);
+      }
+    }
+    generateBookList(bookTitles);
+    */
+
+  const books = {
+    the_nose: { title: 'The Nose', language: 'Arabic (translated from Russian)', author: 'Nikolai Gogol', year: '1836' },
+    the_overcoat: { title: 'The Overcoat', language: 'Arabic (translated from Russian)', author: 'Nikolai Gogol', year: '1842' },
+    the_collected_works: { title: 'The Collected Works', language: 'Arabic (translated from Russian)', author: 'Anton Chekhov' },
+    palace_of_desire: { title: 'Palace of Desire', language: 'Arabic', author: 'Naguib Mahfouz', year: '1957' },
+    sugar_street: { title: 'Sugar Street', language: 'Arabic', author: 'Naguib Mahfouz', year: '1957' },
+    children_of_gebelawi: { title: 'Children of Gebelawi', language: 'Arabic', author: 'Naguib Mahfouz', year: '1959' },
+    the_thief_and_the_dogs: { title: 'The Thief and the Dogs', language: 'Arabic', author: 'Naguib Mahfouz', year: '1961' },
+    autumn_quail: { title: 'Autumn Quail', language: 'Arabic', author: 'Naguib Mahfouz', year: '1962' },
+    a_tale_of_two_cities: { title: 'A tale of Two Cities', language: 'English', author: 'Charles Dickens', year: '1859' },
+    the_happy_prince: { title: 'The Happy Prince', language: 'English', author: 'Oscar Wilde', year: '1888' }
+  };
+
+  function generateBookListsOfInformationWithHeader(obj) {
+    const container = document.getElementById('container');
+    const pageHeader = document.createElement('h1');
+    pageHeader.innerHTML = 'Books I have read';
+    container.appendChild(pageHeader);
+
+    for (const bookName in obj) {
+      const bookSection = document.createElement('div');
+      bookSection.setAttribute('id', bookName);
+      bookSection.setAttribute('class', 'book_sections');
+      container.appendChild(bookSection);
+      const bookObject = obj[bookName];
+
+      for (const property in bookObject) {
+        const bookInformationList = document.createElement('ul');
+        bookSection.appendChild(bookInformationList);
+        if (property === 'title') {
```

Comment:
I really like how you did this.  It shows one of the most important elements of becoming a good programmer: creative laziness. 
Looping through, and setting up your data so it handles the variations you need, is really excellent program design.